lib/omniauth/strategies/apple.rb

```diff
@@ -80,20 +81,22 @@ def id_info
                          verify_aud: true,
                          aud: [options.client_id].concat(options.authorized_client_ids),
                          algorithms: ['RS256'],
-                         jwks: fetch_jwks
+                         jwks: jwks_fetcher
                        }
                        payload, _header = ::JWT.decode(id_token, nil, true, jwt_options)
                        verify_nonce!(payload)
                        payload
                      end
       end
 
-      def fetch_jwks
-        http = Net::HTTP.new('appleid.apple.com', 443)
-        http.use_ssl = true
-        request = Net::HTTP::Get.new('/auth/keys', 'User-Agent' => 'ruby/omniauth-apple')
-        response = http.request(request)
-        JSON.parse(response.body, symbolize_names: true)
+      def jwks_fetcher
```

Comment:
`ruby-jwt` load JWKs w/o `kid` first.
https://github.com/jwt/ruby-jwt/blob/5818f9868ae201f123b2522679bf8af139bbba13/lib/jwt/jwk/key_finder.rb#L26

this isn't what `JSON::JWK::Set::Fetcher` expects.

fill work on the issue later.

Reply:
TODO: replace `ruby-jwt` with `json-jwt` or port JWKS caching feature to `ruby-jwt`